plugins/tagoio-integration/src/front/Auth/Auth.tsx

```diff
@@ -47,9 +57,14 @@ function Auth(props: IAuthProps) {
   const [name, setName] = useState("");
   const [token, setToken] = useState("");
   const [profiles, setProfiles] = useState<any[]>([]);
+  const [selectedRegion, setSelectedRegion] =
+    useState<keyof typeof REGIONS>("us-e1");
 
   const { onSignIn, port } = props;
 
+  // Get the API URL based on selected region
+  const API_URL = REGIONS[selectedRegion].apiUrl;
```

Comment:
`API_URL` name looks like a constant when it isn't, also it's good practice to check when accessing an object if the value exists.

Change to something like:

```ts
const apiURL = useMemo(() => {
  const regionObj = REGIONS[selectedRegion];
  
  return regionObj?.apiUrl || null;
}, [selectedRegion]);
```